dotnet-client-libraries/ITextBasics/ITextBasics/Program.cs

```diff
@@ -0,0 +1,40 @@
+﻿using ITextBasics;
+using ITextBasics.ConsoleManager;
+using ITextBasics.PdfCreatorManager;
+
+var console = new RealConsole();
+var userMenu = new UserMenu(console);
+var pdfCreator = new RealPdfCreator(console);
+var helloWorld = new HelloWorld();
+var bigDocument = new BigDocument();
+
+while (true)
+{
+    userMenu.DisplayOptions();
+    var userAction = userMenu.GetSelection();
+
+    if (userAction == UserMenu.UserAction.Exit)

```

Comment:
This could be moved inside the switch statement

Reply:
Ok, corrected.

But why I programmed it like that in the first place? Because 'Exit' is a special action, out of the ordinary flow, and that is why is in my code outside of other similar actions.

Reply:
It is easier to read and cleaner to place it within the switch statement, since we are already switching over almost all of the other enum states. It makes more sense from a code readability flow to be within the switch. I would argue that the `Unknown` case should also be included explicitly for completeness. To me the "Exit" action is not special, but if you want to make it as a special case and treat it differently, then I think the while loop should become a do..while and the exit condition of the loop should be this if statement (with the opposite check of course). Then you don't need to add that case into the switch and it better captures the intent of the code, and has the benefit of removing a while(true). In general, a while(true) loop with only one exit condition is an anti-pattern, since that style of loop having only one break/exit condition can generally be rewritten as a do..while.

Reply:
i implemented all options in the case statement.